server/objects/Stream.js

```diff
@@ -314,7 +314,7 @@ class Stream extends EventEmitter {
       Logger.info('')
       if (this.isResetting) {
         // AAC encode is much slower
-        const clearIsResettingTime = this.transcodeForceAAC ? 3000 : 500
+        const clearIsResettingTime = this.transcodeForceAAC ? 10000 : 500
```

Comment:
Do you think 10 seconds is necessary here? This means after starting a stream you have to wait 10 seconds before you can seek.

Reply:
the resetting time depends on the power of my machine and my disk speed. My RasPi does need 10s with hdd while on my dev machine needs < 100 ms with ssd. 10s helped on my raspberry pi to stop resetting over and over because he did not finish transcode the first .ts file. File not finished, Path not found => reset loop. My Desktop is fast enough not have the first .ts ready before the frontend request comes in, so he does not fall into the reset loop.

Reply:
Today I did a little research on the "WHY this time is needed" part of this topic. [https://stsaz.github.io/fmedia/audio-formats/](url)
Turns out alac in my case is very slow to decode in comparison to wav or flac and aac slow to encode. On slow devices like the raspi this scales.

If ABS detects an transcode format it defaults to aac. If there is a jump or a bookmark position abs provides a starting point with `-ss` to ffmpeg. Therefor ffmpeg starts decoding 30 seconds before this time stamp. It generates handy `.ts` files to be stramt via hls and generates a `.m3u8` for them. If the transcode starts from the beginning, abs sends out the m3u8  via startloop, startloop handles the connection messages and its good to go. If the client resets the position, the old stream is cleaned up, and a new transcode at a new position is started. There is a case that the transcode has not reached the playing position because of the 30 seconds (slow device), now the client requests this time and because it is not transcoded, the stream rests. This is the resetting loop. In any non resetting case startloop delivered a `.m3u8` file to the client which is used #1625 . THis is contains the chunk info for the audiobook. If a chunk is requested, the info if it is avaliable is not needed. If it is retrieved, other isReset. The client starts to stream if or of not startloop sends ready, because it already happned. With this time we just give the device more time to process the transcode without reset. The client knows where to look for the audio to jump to the position.

Reply:
The socket events for "stream_open" and "stream_ready" aren't actually used for anything anymore so this original implementation you see in `Stream.js` can be misleading. You are mostly correct though but what is most important is the `isResetting` flag.

When the `isResetting` flag is enabled then the stream will not be restarted if a request is made for a segment that doesn't exist yet. This happens on line 48 of `HlsRouter.js`.

The one issue I have with using 10 seconds is that if you start an audiobook and seek to a new position within that 10 second window then the stream is not going to restart. I don't have a good solution for this right now. 

There is more issues with this general approach to HLS streaming with the iOS app. We have no control over how the iOS audio player requests segments and what happens is it will continue requesting segments with no buffer resulting in the stream resetting over and over.

Reply:
This does not sound good. For users in a hurry 3 sec might be to long as well. Did you mean #838?
One possible solution would be to create a stream lock file like output-42.lock file if the first reset comes in. If the second reset try arrives. the lock file exist and the stream is not reset. If the real ts file is finished, and requested, the lockfile is deleted. Maybe a db save with the last resetting chunk number is better. Somehow we need to save this for the next request.
---
Not relevant but during testing I was wondering which format does transcode for which device. alac does transcode for Firefox/Chrome, wma and vorbis does as for some devices others? Right now I am converting alac to flac because since I am not using iTunes anymore there is no reason to keep them. But this is not an valid solution ;)

Reply:
So I quickly build a lockfile solution. It is working on my desktop and my raspi. It does not use the timeout at all, so we do not have to decide on a exact amount of seconds. Seeking now is not blocked at all, it is possible immediately after the stream started. The downside: client tries to get chunk until it exists. This may not the best solution, file system reads, but it seems to work. my working branch: https://github.com/fidoriel/audiobookshelf/tree/lockfilestream